lldb/packages/Python/lldbsuite/test/lldbplatformutil.py

```diff
@@ -34,6 +34,8 @@ def check_first_register_readable(test_case):
         test_case.expect("register read r0", substrs=["r0 = 0x"])
     elif arch in ["powerpc64le"]:
         test_case.expect("register read r0", substrs=["r0 = 0x"])
+    elif arch in ["rv64gc"]:
```

Comment:
Curious here what ` test_case.getArchitecture()` returns, is it exactly this string?

I just remember that riscv can be configured all sorts of ways and I don't know if that ends up in this string.

This is fine for this patch though, folks will find this code easily enough if it's a problem for them.

Reply:
I guess `zero = 0x` would be fine for any `rv(32|64)` target.

Reply:
Yes, you are right. `rv64gc` is just my configuration (now `getArchitecture()` returns what I pass to `--arch` option running  `dotest.py`).

Maybe I can make more generic solution by adding a regexp or just simple `if arch.startswith("rv"): arch = "riscv"` in `getArchitecture()`. I think it would be enough for now, because `rv32` and `rv64` have the same register set and as you have noticed `zero = 0x` would be fine for any target, so there is no reasons to distinguish a bitness or configurations.

Reply:
To be honest I was asking more for understanding than anything else. Good to know it would be easy to extend.

I would leave this as is on the grounds that you only know for sure that rv64 works here, and anyone working on anything else will have the ability to validate the test themselves. So leave it to future them (maybe future you) to update the name here.

Just update the PR title and commit message to be clear that you're doing this for rv64 only.

Reply:
If you have a way to test rv32 as well that's great but up to you.

Reply:
I have finally decided to add simple regexp, so it's going to work with any riscv target, not only `rv64gc`.